### PR TITLE
[WP4] Add ONNX export functionality for Simplified HRM models

### LINKED ISSUE #65: [WP4] Export trained Maze model to ONNX
### Description

Export the trained maze model to ONNX. The maze pipeline differs from
Sudoku in two key ways: the output head produces binary (2-class) logits
instead of 10-class, and prediction feedback is disabled (input/output
are different token domains). The ONNX wrapper already handles both.

### Tasks

- [ ] Export maze model: `python scripts/export_onnx.py --model model/simplified_hrm_maze_best.pt --puzzle maze`
- [ ] Export step-count variants (16, 8, 4)
- [ ] Verify ONNX output matches PyTorch predictions
- [ ] Run `solve_onnx.py --puzzle maze --benchmark` on x86 to get baseline latency

### Acceptance Criteria

- [ ] ONNX export succeeds without errors
- [ ] Verification passes (max diff < 1e-5)
- [ ] Maze-specific metrics work in `solve_onnx.py` (cell accuracy, path accuracy)

### Commands

```bash
python scripts/export_onnx.py --model model/simplified_hrm_maze_best.pt --puzzle maze --output model/simplified_hrm_maze_s16.onnx
python scripts/export_onnx.py --model model/simplified_hrm_maze_best.pt --puzzle maze --reasoning-steps 8 --output model/simplified_hrm_maze_s8.onnx
python scripts/export_onnx.py --model model/simplified_hrm_maze_best.pt --puzzle maze --reasoning-steps 4 --output model/simplified_hrm_maze_s4.onnx
```

### PR DESCRIPTION
**Pull request by KyryloKozlovskyi**
[WP4] Export trained Maze model to ONNX https://github.com/fionntmcc/cross-platform-hrm/issues/65

### CODE REVIEW

**Review by fionntmcc — APPROVED** [review]
Very cool, good work.
